src/index.js

```diff
@@ -1,14 +1,27 @@
 const beginFloat = '~begin~float~';
 const endFloat = '~end~float~';
 
-const StringifyWithFloats = (config = {}) => (value, options) => {
-  const jsonReplacer = (key, value) => {
-    const shouldBeReplaced = config[key] === 'float'
+const StringifyWithFloats = (config = {}) => (inputValue, inputReplacer, space) => {
+  let isFirstIteration = true;
+  const jsonReplacer = (key, val) => {
+    if (isFirstIteration) {
+      isFirstIteration = false;
+      return (typeof inputReplacer === 'function') ? inputReplacer(key, val) : val;
+    }
+    let value;
+    if (typeof inputReplacer === 'function') {
+      value = inputReplacer(key, val);
+    } else if (Array.isArray(inputReplacer)) {
+      value = (inputReplacer.indexOf(key) !== -1) ? val : undefined;
```

Comment:
I don't understand what is happening here with `(Array.isArray(inputReplacer)` and ` (inputReplacer.indexOf(key) !== -1) ? val : undefined` -- can you explain?

Reply:
JSON.stringify's [`replacer`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#The_replacer_parameter) parameter can be a function or an array specifying which properties to keep in the object being stringified.

Since we're always passing in a `replacer` function to JSON.stringify, we have to implement the logic when `replacer` is an array. So, this returns `undefined` which removes the property if the key is not contained in the `replacer` array.